services/report.py

```diff
@@ -61,42 +51,20 @@ def build_report(chunks, files, sessions, totals, report_class=None):
 
 
 @sentry_sdk.trace
-def build_report_from_commit(commit: Commit, report_class=None):
+def build_report_from_commit(commit: Commit, report_class=None) -> Report | None:
```

Comment:
wait why aren't we just using `shared.reports.api_report_service`'s copy of this (which should maybe be renamed `shared.reports.django_report_service`)? is that an @adrian-codecov question?

Reply:
That is a good question, and I believe reducing the code duplication by removing one copy completely in favor of another is a good idea.

When working of this, I was bashing my head against the table quite a bit, I was even thinking that the definitions for `shared.core` were somehow shadowing `core`, even though I have no idea how that would be even possible.

This is good feedback, and I might just go ahead and remove this from api completely in favor of the definition in `shared`.